syllcorpus.m

```diff
@@ -1,206 +1,186 @@
-%  UNTITLED2 Summary of this function goes here
-%   Detailed explanation goes here
 
-% Default argument  
-if nargin < 7
-    disp('Using default arguments.')
+
+function syllcorpus(extract,extractNext,alifile,wavscp,model,phones,transcript,obase)
+  %  Outputs extractions from the given transcript to files with prefix
+  %  obase. Determines which words to extract via extract
+  %
+  % INPUT:
+  % extract: function from cell array of string, representing phones of a
+  %          word, to whether that word should be extracted  
+  % extractNext : bool - whether the word following a word that satisifies
+  %                      extract should also be extracted
+
+  % Default arguments
+  if nargin < 1
+    extract = @bisyllable; % if relying on default, you should check that num_vowels 
+                           % defined in bisyllable is what you want
+  end 
+  if nargin < 2
+    extractNext = false;
+  end
+  if nargin < 7
+    disp('Using default arguments.');
     alifile = '/Volumes/NONAME/speech/librispeech/s5/exp/tri4b/ali-e3-t.gz';
     wavscp = '/Volumes/NONAME/speech/librispeech/s5/data/train_clean_100/wav-e3.scp';
     model = '/Volumes/NONAME/speech/librispeech/s5/exp/tri4b/final.mdl';
     phones = '/Volumes/NONAME/speech/librispeech/s5/data/lang_nosp/phones.txt';
     transcript = '/Volumes/NONAME/speech/librispeech/s5/data/train_clean_100/text-e3';
-    obase = '/local/matlab/Kaldi-alignments-matlab/data/tri1b';
-    nvowel = 3;
-end
+    obase = '/local/matlab/Kaldi-alignments-matlab/data/tri1b';    
+  end
 
-% Read wav file and alignment for all the utterance IDs.
-% Map from uid to wav launch pipes.
-Scp = load_kaldi_wavscp(wavscp);
-
-% Read transcript.
-Tra = load_kaldi_transcript(transcript);
-
-% For mapping back and forth between phones and their indices.
-P = phone_indexer(phones);
-
-% Create and load the alignments in various formats.
-% Cell array of Uid, and cell array of alignment vectors.
-[Uid,Basic,Align_pdf,Align_phone,Align_phone_len] = load_ali(alifile,model);
-
-% Index in Uid and Align of the utterance being displayed.
-ui = 1;
-
-% Maximum value for ui.
-[~,U] = size(Uid);
-
-% Initialize some variables that are set in nested functions.
-uid = 0; uid2 = 0; F = 0; Sb = 0; Pb = 0; Wb = 0; w = 0; fs = 0;
-
-M = 0; S1 = 0; SN = 0; N = 100;
-F = 0; F1 = 0; FN = 0; nsample = 0; nframe = 0; 
-PX = 0; ya = 0; tra = 0; wi = 1;
-Fn = 0;
-
-% Make flac available
-setenv('PATH', '/opt/local/bin:/opt/local/sbin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin');
-% setenv('DYLD_LIBARARY_PATH', '/opt/local/lib:/usr/local/lib:/usr/lib');
-% Run display program.
-system('which flac');
-
-utterance_data(ui);
- 
-
-% [PH,SU,PHstart,PHend,SUstart,SUend,WRstart,tra] = parse_ali2(uid,Align_pdf,Align_phone,Tra,P,n)
-
-% Set phone and audio data for k'th utterance.
-% Values are for utterance k.
-    function utterance_data(k)
```

Comment:
Many of the variables set by this function were never used, and after eliminating the void lines the function became smaller. I've inlined inside the below for loop (where utterance_data was originally called).